3_javascript/javascript1/week2/javascript.js

```diff
@@ -0,0 +1,102 @@
+// Formal fullname
+function getFullname(firstname, surname, useFormalName, isWoman) {
+    if (firstname === "" || surname === "") {
+        return `Please enter your full name`;
+    }
+
+    if (useFormalName === true && isWoman === true) {
+        return `Lady ${firstname} ${surname}`;
+    }
+    else if (useFormalName === true && isWoman === false) {
+        return `Lord ${firstname} ${surname}`;
+    }
+    else {
+        return `${firstname} ${surname}`;
+    }
+}
+
+const fullName1 = getFullname("Benjamin", "Hughes", true, false);
+const fullName2 = getFullname("Benjamin", "Hughes", false, false); 
+const fullName3 = getFullname("", "Wardak", false, false);
+const fullName4 = getFullname("Delilah", "Wardak", true, true);
+const fullName5 = getFullname("", "", false, false);
+
+console.log(fullName1);
+console.log(fullName2);
+console.log(fullName3);
+console.log(fullName4);
+console.log(fullName5);
+
+// Event application
+
+function getEventWeekday(numberOfTheDay){
+    let today = new Date().getDay();
+    let weekdays = ["Sunday", "Monday", "Tuesday", "Wednesday" , "Thursday", "Friday", "Saturday"];
+   
+    return weekdays[(today + numberOfTheDay) % 7];
+}
+
+console.log(getEventWeekday(0));
+console.log(getEventWeekday(1));
+
+// Weather wear
+// Create a function (that you have to name) that has temperature as parameter. 
+// Based on the temperature it should return a string with what the user should wear. 
+// You decide what the user should wear based on the temperature.
+function toWear (temperature) {
+let clothes = "";
+if (temperature <=0) {
+    clothes = `Steal some clothes from eskimos, you are freezing to death!`; 
+}
+else if (temperature > 0 && temperature < 15) {
+    clothes = `You can wear whatever you like`;
+}
+else {
+    clothes = `Time to strip down to shorts and a t-shirt`;
+}
+return clothes;
+
+}
+let clothesType = toWear(-2);
+console.log(clothesType);
+
+
+
+// Student manager
+const class07Students = [];
+
+function addStudentToClass(studentName) {
+    if (class07Students.includes(studentName)) {
+        console.log(`Student ${studentName} is already in the class.`);
+    } else {
+        if (class07Students.length < 7 || class07Students.length > 6 && studentName === "Queen") {
```

Comment:
it is a good practice to use () to group together the conditions that belong together. (in this case the second 2).
This will still work, because in javascript the logical AND operator (&&) has higher precedence than the logical OR operator (||). This means that the expression inside the && is evaluated before the ||. But there could be cases where not grouping them will cause a different behavior.

Reply:
I had a little issue with when I had "Queen" as a seperate condition. It was returning results even though the number of students was greater than 7. i then put it like this and then it started to behave :)